.ado/templates/build-rnw.yml

```diff
@@ -31,7 +31,8 @@ steps:
       installVsComponents: ${{ parameters.installVsComponents }}
       debug: ${{ parameters.debug }}
 
-  - task: NuGetCommand@2
+  # NuGetCommand workaround: https://developercommunity.visualstudio.com/content/problem/288534/vsts-yaml-build-failure-the-task-name-nugetcommand.html
+  - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2
```

Comment:
The VSBuild task below has a "restoreNuGetPackages" parameter. Consider removing this task and passing the param to just let VS do the restore.

Reply:
I don't remember the history as to why we do the nuget commands outside of the VSTask, so I think it's out of scope for this PR.

Reply:
I don't think there's a history, hit there's a difference in behavior. The restore task uses the environment version of NuGet CLI instead of version attached to VS. This has caused problems for us before.

Summary is VS restore is likely better for our purposes. Would ask that you try it, to avoid the workaround this PR adds.

Reply:
The CLI, our CLI, calls nuget restore before calling msbuild, rather than use VS's restore capability, right? I think this keeps us in alignment. But I can try it out.

Reply:
As of 0.64, our CLI uses MSBuild built-in functionality to do the NuGet restore instead of using the NuGet CLI.

Previous strategy of using the CLI led to issues with version mismatch (updates can require new NuGet versions) and required some special distribution work.

Reply:
Hmm, apparently the parameter is deprecated: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/visual-studio-build?view=azure-devops

> This option is deprecated. Make sure to clear this checkbox and instead use the NuGet Installer build task.

Reply:
Oh huh. Thanks for looking at that. Seems fine to use the NuGet task + explicit guid then.